iceoryx_dds/source/iceoryx_dds/dds/cyclone_data_writer.cpp

```diff
@@ -48,10 +50,45 @@ void iox::dds::CycloneDataWriter::connect() noexcept
     LogDebug() << "[CycloneDataWriter] Connected to topic: " << topic;
 }
 
-void iox::dds::CycloneDataWriter::write(const uint8_t* const bytes, const uint64_t size) noexcept
+void iox::dds::CycloneDataWriter::write(iox::dds::IoxChunkDatagramHeader datagramHeader,
```

Comment:
This is weird and somehow wrong, we have here a method which can fail for various reason but never informs the user and from the API point of view it seems like this method always succeeds.
Would it be possible to return an expected here which returns the `LogError()` message in a descriptive enum value?

Reply:
Not part of this PR and should be done by #376

Reply:
added to the issue